com.unity.render-pipelines.core/Runtime/Common/CoreAttributes.cs

```diff
@@ -21,4 +21,26 @@ public class DisplayInfoAttribute : Attribute
     public class AdditionalPropertyAttribute : Attribute
     {
     }
+
+    /// <summary>
+    /// Attribute to specify that a class is additional data of another component
+    /// Currently is being used for HDRP/URP additional Camera and Light data
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public class AdditionalComponentData : Attribute
+    {
+        /// <summary>
+        /// Summary the component type of is additional data
+        /// </summary>
+        public Type componentType;
+
+        /// <summary>
+        /// Constructor of the attribute
+        /// </summary>
+        /// <param name="componentType">The component type</param>
+        public AdditionalComponentData(Type componentType)
+        {
+            this.componentType = componentType;
+        }
+    }
```

Comment:
We didn't really need them as we previously considered that this was an editor workflow to manage. So we ended with having interface but on Editor side (such `IRemoveAdditionalDataContextualMenu<Camera>`).

And the plan was to remove the additional data sibling component. So when we will be able to finish & land this, the attribute introduced here (or the former interface) will don't have any usage. That is why even if the need for it increased last year, I didn't add it as it add a public API that should be deprecated soon. :)

Now as the additional data refactor is just so overwhelming and it is a nightmare to know exactly when this will be finished, I'm ok with this introduction. But as it is public API, some user can want to use it in their custom SRP with component that we have not think of yet.  Should we ensure that if this happens, can we ensure our system will works in this case?

Also, HD have additional component for Reflection Probe too (regarding comments). There is no conflict as URP don't have additional data for this so there is also no contextual menu dispatcher. But I would prefer to unify our workflow and thus use this component also for HDReflectionProbe.

Also as we partly automate the concept of additional data here, as a user, I would expect to not have to write RemoveAdditionalComponent contextual menu command and such when I wrote my own SRP. But this seams tricky to automate...

Lastly, I wonder if this should inherite `[RequireComponent(typeof(Camera))]` as it seams to me we always have this one when using the `[AdditionalComponentData(typeof(Camera))]`.

Reply:
I will have a look at the Reflection Probe and how to automate the context menu. 

The RequierComponent is sealed, so we can not inherit from it.

Reply:
Since we also discussed inheriting from RequireComponent, I wonder if we could make it non-sealed? I'm not sure if there's a good reason to keep it sealed.